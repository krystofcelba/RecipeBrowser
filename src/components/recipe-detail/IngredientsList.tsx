import React from 'react';
import glamorous from 'glamorous-native';
const { View } = glamorous;

import i18n from 'src/assets/i18n';
import { Ingredient } from 'src/services/api';
import { SubTitleText, BoldText, LeftPaddedText } from 'src/components/common';

interface Props {
  ingredients: [Ingredient];
  seasonings: [string];
}

const IngredientsList = ({ ingredients, seasonings }: Props) => (
  <View flex={1}>
    <SubTitleText>{i18n.t('ingredients')}</SubTitleText>
    {ingredients.map(({ amount, unit, name }, i) => (
      <View flexDirection="row" key={i}>
        <LeftPaddedText>{'\u2022'}</LeftPaddedText>
        <LeftPaddedText>{`${amount} ${unit} ${name}`}</LeftPaddedText>
      </View>
    ))}
    <View flexDirection="column">
      <BoldText>Seasonings</BoldText>
      <LeftPaddedText>
        {seasonings.map((seasoning, i) => `${seasoning}${i < seasonings.length - 1 ? ', ' : ''}`)}
      </LeftPaddedText>
    </View>
  </View>
);

export default IngredientsList;
