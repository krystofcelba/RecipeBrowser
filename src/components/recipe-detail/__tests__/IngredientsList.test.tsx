import React from 'react';
import { shallow } from 'enzyme';
import IngredientsList from '../IngredientsList';

const mockedRecipes = require('../../../services/__mockData__/recipes.json');

describe('IngredientsList component', () => {
  const mockedRecipe = mockedRecipes[2];
  it('renders as expected', () => {
    const wrapper = shallow(
      <IngredientsList ingredients={mockedRecipe.ingredients} seasonings={mockedRecipe.seasonings} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
