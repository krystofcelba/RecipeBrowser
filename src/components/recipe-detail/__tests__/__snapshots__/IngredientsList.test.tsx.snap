// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientsList component renders as expected 1`] = `
<glamorous.View
  flex={1}
>
  <SubTitleText>
    Ingredients
  </SubTitleText>
  <glamorous.View
    flexDirection="row"
    key="0"
  >
    <LeftPaddedText>
      •
    </LeftPaddedText>
    <LeftPaddedText>
      200 g white fish fillet (cod, plaice, sole; fresh or frozen)
    </LeftPaddedText>
  </glamorous.View>
  <glamorous.View
    flexDirection="row"
    key="1"
  >
    <LeftPaddedText>
      •
    </LeftPaddedText>
    <LeftPaddedText>
      0.5  avocados
    </LeftPaddedText>
  </glamorous.View>
  <glamorous.View
    flexDirection="row"
    key="2"
  >
    <LeftPaddedText>
      •
    </LeftPaddedText>
    <LeftPaddedText>
      0.5  grapefruits
    </LeftPaddedText>
  </glamorous.View>
  <glamorous.View
    flexDirection="row"
    key="3"
  >
    <LeftPaddedText>
      •
    </LeftPaddedText>
    <LeftPaddedText>
      1  chopped jalapeño
    </LeftPaddedText>
  </glamorous.View>
  <glamorous.View
    flexDirection="row"
    key="4"
  >
    <LeftPaddedText>
      •
    </LeftPaddedText>
    <LeftPaddedText>
      0.5 bunches coriander
    </LeftPaddedText>
  </glamorous.View>
  <glamorous.View
    flexDirection="row"
    key="5"
  >
    <LeftPaddedText>
      •
    </LeftPaddedText>
    <LeftPaddedText>
      0.5  limes (juice)
    </LeftPaddedText>
  </glamorous.View>
  <glamorous.View
    flexDirection="row"
    key="6"
  >
    <LeftPaddedText>
      •
    </LeftPaddedText>
    <LeftPaddedText>
      100 g chopped cabbage
    </LeftPaddedText>
  </glamorous.View>
  <glamorous.View
    flexDirection="row"
    key="7"
  >
    <LeftPaddedText>
      •
    </LeftPaddedText>
    <LeftPaddedText>
      1 tsp coconut oil
    </LeftPaddedText>
  </glamorous.View>
  <glamorous.View
    flexDirection="column"
  >
    <BoldText>
      Seasonings
    </BoldText>
    <LeftPaddedText>
      Chili flakes, 
      unrefined salt, 
      black pepper
    </LeftPaddedText>
  </glamorous.View>
</glamorous.View>
`;
